File: Oauth2Simples/src/main/java/br/com/estudos/oauth2/model/Balance.java
package br.com.estudos.oauth2.model;

import javax.persistence.Entity;
import javax.persistence.Column;
import javax.persistence.Id;
import javax.persistence.GenerationType;
import javax.persistence.GeneratedValue;
import javax.persistence.ManyToOne;
import javax.persistence.JoinColumn;
import org.hibernate.annotations.Type;

import java.math.BigDecimal;
import java.util.UUID;

@Entity
public class Balance{

    @Id
    @GeneratedValue
    @Type(type = "uuid-char")
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "id_type_balance")
    private TypeBalance typeBalance;

    @Column
    private BigDecimal value;

    public UUID getId() {
        return this.id;
    }

    public TypeBalance getTypeBalance() {
        return this.typeBalance;
    }

    public void setTypeBalance(TypeBalance typeBalance) {
        this.typeBalance = typeBalance;
    }

    public BigDecimal getValue() {
        return this.value;
    }

    public void setValue(BigDecimal value) {
        this.value = value;
    }




}

File: kotlin/kotlinEstudos/src/main/java/br/com/kotlin/estudos/model/repository/financial/Balance.kt
package br.com.kotlin.estudos.model.repository.financial

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import java.util.*
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.type.SqlTypes
import java.math.BigDecimal
import java.sql.Date

@Entity
@Table(name="BALANCE")
class Balance(
     @Id
     @Column(nullable = false)
     @JdbcTypeCode(SqlTypes.CHAR)
     val id: UUID = UUID.randomUUID(),

     @JsonIgnore
     @ManyToOne(fetch = FetchType.LAZY)
     @JoinColumn(name="usuario_id")
     @JdbcTypeCode(SqlTypes.CHAR)
     var user: Usuario?,

     @ManyToOne(fetch = FetchType.LAZY)
     @JoinColumn(name="type", referencedColumnName = "type")
     var typeBalance: TypeBalance?,

     @Column(name = "VALUE_CONSUME")
     var valueConsume: BigDecimal,

     @Column(name = "DATE_CONSUME")
     var dateConsume: Date,

     @Column(name = "DESCRIPTION")
     var description: String

     ){}